src/v/redpanda/admin_server.cc

```diff
@@ -1403,19 +1403,19 @@ admin_server::raft_transfer_leadership_handler(
 
     co_return co_await _partition_manager.invoke_on(
       shard,
-      ss::coroutine::lambda(
         [group_id, target, this, req = std::move(req)](
-          cluster::partition_manager& pm) mutable
-        -> ss::future<ss::json::json_return_type> {
+          cluster::partition_manager& pm) mutable {
             auto partition = pm.partition_for(group_id);
             if (!partition) {
                 throw ss::httpd::not_found_exception();
```

Comment:
The correct translation is to replace `throw` with `return ss::make_exception_future()`, but as long as we're happy that no intermediate exception handling is necessary, this is ok.